### PR TITLE
fix: Algebraic effects in REPL expressions

### LINKED ISSUE #8601: No check for unhandled effects on entrypoint
```
pub def main(): Unit \ Console = Console.println("hey")
```
Running in vscode (`:eval main()`) crashes at runtime

### PR DESCRIPTION
**Pull request by JonathanStarup**
Fixes #8601

This PR correctly checks for algebraic effects, but also disallows any non-primitive effect. I think this is okay, because how would you even use regions in the repl?

@magnus-madsen Env is missing in Prelude - do you have a documentation comment in mind that I can insert?

### CODE REVIEW

**Comment by magnus-madsen** [comment]
>@magnus-madsen Env is missing in Prelude - do you have a documentation comment in mind that I can insert?

Something about accessing properties of the system?

File: main/src/library/Prelude.flix [hunk 1]
```diff
@@ -17,6 +17,15 @@ pub enum Proxy[_] {
     case Proxy
 }
 
+///
+/// The uninterpretable `Env` effect.
+///
+/// The `Env` effect represents ..
```
Comment (magnus-madsen) [inline main/src/library/Prelude.flix:23]: ... actions that read from the environment (e.g. information about the shell, operating system, or hardware)

**Comment by magnus-madsen** [comment]
@JonathanStarup Ping

**Comment by JonathanStarup** [comment]
@magnus-madsen sorry, got it done now

**Pull request merged**